tiledb/sm/c_api/tiledb.cc

```diff
@@ -3375,6 +3375,7 @@ int32_t tiledb_array_open(
   return TILEDB_OK;
 }
 
+/* This is a deprecated API */
```

Comment:
We only need to document deprecated APIs in the header. Remove from the implementation file.